cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/LocalMetricsCollectionService.java

```diff
@@ -129,18 +136,23 @@ protected void shutDown() throws Exception {
 
   /**
    * Creates a task for cleanup.
-   * @param retention Retention in seconds.
+   *
+   * @param secondRetention Retention in seconds for second resolution table.
+   * @param minRetention Retention in seconds for min resolution table.
+   * @param hourRetention Retention in seconds for hour resolution table.
    */
-  private Runnable createCleanupTask(final long retention) {
+  private Runnable createCleanupTask(long secondRetention, long minRetention, long hourRetention) {
     return new Runnable() {
       @Override
       public void run() {
         // We perform CleanUp only in LocalMetricsCollectionService , where TTL is NOT supported
         // by underlying data store.
         long currentTime = TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS);
-        long deleteBefore = currentTime - retention;
         try {
-          metricStore.deleteBefore(deleteBefore);
+          // delete metrics from corresponding metrics resolution table
+          metricStore.deleteBefore(currentTime - secondRetention, Constants.Metrics.SECOND_RESOLUTION);
+          metricStore.deleteBefore(currentTime - minRetention, Constants.Metrics.MINUTE_RESOLUTION);
+          metricStore.deleteBefore(currentTime - hourRetention, Constants.Metrics.HOUR_RESOLUTION);
         } catch (Exception e) {
           throw Throwables.propagate(e);
         }
```

Comment:
Can't comment on the next line. But the next time this task should executes as the smaller retention interval.

Reply:
Fixed